### ISSUE TITLE
System to keep aggregates separate by report type

### ISSUE THREAD

**Issue by AureliaKhorsand** [fecgov/fecfile-web-app#1580]
Business Reason

System should ensure that aggregates are calculated only within the same report type (e.g., Form 3X transactions are aggregated with Form 3X transactions, but NOT Form 3 transactions) to prevent double counting.

Acceptance Criteria

**Given** an aggregated transaction
**When** the transaction is saved
**Then** the aggregate will ONLY be calculated within the same report type.

### DEV Notes ###
null

### Design ###
null

 See full ticket and images here: [FECFILE-531](https://fecgov.atlassian.net/browse/FECFILE-531)

**Comment by AureliaKhorsand** [comment]
@toddlees we briefly referenced this recently - can you or @mjtravers comment on current system functionality? I am assuming that aggregates are calculated by entity (or per election office sought) and aggregate group, but there are no guard rails on report type. Is that right?

**Comment by mjtravers** [comment]
@AureliaKhorsand Yes, that's correct. The aggregate value for a primary contact includes all transactions that the contact is associated with as the primary without taking into consideration the report type.

**Comment by toddlees** [comment]
@AureliaKhorsand @mjtravers I think one thing we determined was that, while a transaction in an f3x never includes f24 transactions in it's aggregate, a transaction in an f24 would still need to include past f3x transactions in it's aggregate.  

<img width="1112" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNXRSQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--80704f9f4738b32f31b0d3ef3f06c4f5d7a4cfca/image.png" alt="image.png" />

**Comment by toddlees** [comment]
@AureliaKhorsand  @mjtravers I'm not sure this ticket is relevant any longer given that a transaction on F24 is the same record as that transaction on the F3X.

**Comment by AureliaKhorsand** [comment]
Confirmed with dev team that this ticket is still relevant as this is a universal rule for _all_ reports (see requirements notes on 12/19/23).

**Comment by AureliaKhorsand** [comment]
On hold per Ryan's concerns on Slack. Needs further discussions with Ryan.

**Comment by MitchellTCG** [comment]
@AureliaKhorsand 
I moved this back to AA until a decision is made re: Ryan's comments

**Comment by GreggMoreland** [comment]
@AureliaKhorsand where might I find Ryan's comments?
@mjtravers @toddlees this ticket will likely be needed once we start on the other reports (3, 3P) which calculate aggregates based on election cycle, not on calendar year....so an individual aggregate for 3x would be different than the aggregate for 3.  Any thoughts or comments on this?

**Comment by toddlees** [comment]
@GreggMoreland @AureliaKhorsand @mjtravers agreed, the entity aggregates over an election cycle will have to be handled independently.  I'm still surprised by the comment in the 12/19/23 meeting:
> Aggregates carry over from report to report, but do not get combined between report types (e.g., between Form 3X and Form 3).

Certainly entity aggregates over and election period (form 3) and calendar year (form 3x) would be calculated separately; they are over different windows of time.   However, I'm having trouble understanding why we would not want the dollar amount of a transaction in an f3x to factor in to the entity election-cycle aggregate of an f3.

Edit:
I guess what I'm wondering is this statement to be interpreted as the aggregates don't interact (i wholeheartedly agree) o r that the transactions are not referenced across report types (i have suspicions about this approach)

**Comment by AureliaKhorsand** [comment]
This has been confirmed as NOT NEEDED for test release 1, since this won't be relevant until we have additional report types. This ticket also needs more clarification from FEC on how aggregates will work if a committee changes type and has reports that track aggregates differently. Moving to Product Backlog until a later date.
